### PR TITLE
Examples for Helmholtz_Neumann problem for both coskx and eikx cases

### LINKED ISSUE #7: Neumann Complex with coskx
Case:
- Helmholtz problem for a 2D square with Neumann null boundary conditions.
- Set `f= k^2 * [cos(kx),cos(kx)]`. The analytic solution is `u = [cos(kx), cos(kx)]`.

To do: 

- [x] Validate [script](https://github.com/pescap/ScattPINNs/blob/main/examples/forward/Helmholtz_Neumann_2d_complex_coskx.py) 
- [x] Validate [unit tests](https://github.com/pescap/ScattPINNs/actions) for the script 
- [ ] Validate the [doc](https://github.com/pescap/ScattPINNs/blob/main/docs/source/forward/helmholtz.2d.neumann.complex.coskx.rst) for the script.

### LINKED ISSUE #8: Neumann complex with eikx
Case:

Helmholtz problem for a 2D square with Neumann non null boundary conditions.
Set f= k^2 * [cos(kx),sin(kx)]. The analytic solution is u = [cos(kx), sin(kx)].
To do:

- [ ] Validate [script](https://github.com/pescap/ScattPINNs/blob/main/examples/forward/Helmholtz_Neumann_2d_complex_coskx.py)
- [ ] Validate [unit tests](https://github.com/pescap/ScattPINNs/actions) for the script
- [ ] Validate the [doc](https://github.com/pescap/ScattPINNs/blob/main/docs/source/forward/helmholtz.2d.neumann.complex.coskx.rst) for the script.

Help: Check the scripts with `eikx` in [dev](https://github.com/pescap/ScattPINNs/tree/main/dev/waves-deepxde)

### PR DESCRIPTION
**Pull request by Ehaw04**
Here´s the implementation and some test for both examples, missing the docs

### CODE REVIEW

**Comment by Ehaw04** [comment]
@pescap Travis CI seems to be working fine for my PR, I will fix the Codacy marked errors (there are some white spaces in my code)

File: examples/forward/Helmholtz_Neumann_2d_complex_eikx.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+def pde(x, y):
+    yRe, yIm = y[:, 0:1], y[:, 1:2]
+    dyRe_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)
+    dyRe_yy = dde.grad.hessian(y, x, component=0, i=1, j=1)
+    dyIm_xx = dde.grad.hessian(y, x, component=1, i=0, j=0)
+    dyIm_yy = dde.grad.hessian(y, x, component=1, i=1, j=1)
+    #Projection to the real and imaginary axes
+    fRe = k0 ** 2 * cos(k0 * x[:, 0:1]) * cos(k0 * x[:, 1:2])
+    fIm = k0 ** 2 * sin(k0 * x[:, 0:1]) * sin(k0 * x[:, 1:2])
+
```
Comment (pescap) [inline examples/forward/Helmholtz_Neumann_2d_complex_eikx.py]:
```suggestion
```

**Comment by pescap** [comment]
The Github Actions workflows for _cosk and _eikx last too long. We should reduce the number of epochs.

File: examples/forward/Helmholtz_Neumann_2d_complex_coskx.py [hunk 2]
```diff
@@ -6,17 +6,10 @@
 n = 2
 precision_train = 10
 precision_test = 10
-hard_constraint = True
-weights = 100  # if hard_constraint == False
-epochs = 5000 # tested with 50.000 epochs.
-parameters = [1e-3, 4, 80, "sin"]
+weights = 100
+epochs = 50000 # tested with 50.000 epochs.
```
Comment (pescap) [inline examples/forward/Helmholtz_Neumann_2d_complex_coskx.py]:
```suggestion
epochs = 5000 # tested with 50.000 epochs.
```

File: examples/forward/Helmholtz_Neumann_2d_complex_eikx.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import deepxde as dde
+import numpy as np
+
+# General parameters
+n = 2
+precision_train = 10
+precision_test = 10
+weights = 100
+epochs = 50000 # tested with 50.000 epochs.
```
Comment (pescap) [inline examples/forward/Helmholtz_Neumann_2d_complex_eikx.py]:
```suggestion
epochs = 5000 # tested with 50.000 epochs.
```

**Pull request merged**